src/Cmd/Extra.elm

```diff
@@ -164,3 +164,31 @@ withTrigger msg =
 addTrigger : msg -> ( model, Cmd msg ) -> ( model, Cmd msg )
 addTrigger msg =
     add <| perform msg
+
+
+{-| Add new cmd to an existing pair under a certain condition.
+
+    prevCmd : Cmd String
+    prevCmd =
+        perform "foo"
+
+    newCmd : Cmd String
+    newCmd =
+        perform "bar"
+
+    ( "model", prevCmd )
+      |> addIf False newCmd
+      |> Tuple.second
+      |> ((==) prevCmd)
+    --> True
+
+-}
+addIf : Bool -> Cmd msg -> ( model, Cmd msg ) -> ( model, Cmd msg )
+addIf predicate newCmd ( model, prevCmd ) =
```

Comment:
for some reason I think `addWhen` would be a better name but I'm not sure if I can justify it. Maybe because I feel every `if` has to has `else`?

Reply:
Yes, that is true.

My reasons:
- I was following [Html.Extra.viewIf](https://package.elm-lang.org/packages/elm-community/html-extra/latest/Html-Extra#viewIf) naming "convention"
- `if` is closer to the computer language, so as a package user I immediately understand what the intention of `if`. However `when` is somewhat new, I have to look at the algorithm to be sure what is that exactly about.

Reply:
html-extra package in particular is not really the best quality one. I'm for instance using https://package.elm-lang.org/packages/turboMaCk/glue/latest/Glue#subscriptionsWhen `when` in Glue package myself.

I think neither is close to computer language. It might be more similar name to `if` as a language construct but this is a bit missleading because:

- in pure language every if must have else producing the same type
- with if the expressions are evaluated lazily. if the result of condition is `False` then in never evaluated. But this case is a functions and in elm arguments are evaluated strictly so the when branch is always avaluated, just not used in some case.
- in pure FP when is usually used "single branch" conditions `when :: Applicative f => Bool -> f () -> f ()` https://hoogle.haskell.org/?hoogle=when&scope=set%3Astackage

As I said I don't really have super strong argument (or I don't really belive those are strong enough) but I feel they are still stronger than `if` being like `if` which it clearly isn't.